morango/controller.py

```diff
@@ -73,24 +78,27 @@ def serialize_into_store(self):
             Store.objects.filter(id__in=deleted_ids).update(deleted=True)
             DeletedModels.objects.all().delete()
 
-    @transaction.atomic
-    def store_to_app(self):
+    def deserialize_from_store(self):
         """
         Takes data from the store and integrates into the application.
         """
-        syncable_dict = _profile_models[self.profile]
-        # iterate through classes which are in foreign key dependency order
-        for model_name, klass_model in iteritems(syncable_dict):
-            for store_model in Store.objects.filter(model_name=model_name, dirty_bit=True):
-                if store_model.deleted:
-                    klass_model.objects.filter(id=store_model.id).delete()
-                else:
-                    app_model = klass_model.deserialize(json.loads(store_model.serialized))
-                    try:
-                        app_model.save(update_dirty_bit_to=False)
-                    # when deserializing, we catch this exception in case we have a reference to a missing object (foreign key) not in the app layer
-                    except ObjectDoesNotExist:
-                        DeletedModels.objects.update_or_create(defaults={'id': app_model.id, 'profile': app_model.morango_profile},
-                                                               id=app_model.id)
-                store_model.dirty_bit = False
-                store_model.save(update_fields=['dirty_bit'])
+
+        self.serialize_into_store()
```

Comment:
I think we need this to happen within the `transaction.atomic()`, to ensure nothing gets modified in the app between serialization and deserialization.